### PR TITLE
Add BambooHR connector

### PR DESCRIPTION
**Pull request by huongtn**
## Description

> Provide description here

## Type of change
- [ ] Bug fix
- [ #435 ] New feature
- [ ] Breaking change

## Related issues

> Fix [#435 ]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: sources/bamboohr-source/test/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  test('check connection - no client_secret', async () => {
+    const source = new sut.BambooHRSource(logger);
+    await expect(
+      source.checkConnection({
+        api_key: '',
+        domain: '',
+      } as any)
+    ).resolves.toStrictEqual([
+      false,
+      new VError('api_key must be a not empty string'),
```
Comment (willmarks) [inline sources/bamboohr-source/test/index.test.ts:54]: `api_key cannot be an empty string`

Comment (willmarks) [inline sources/bamboohr-source/test/index.test.ts:54]: `api_key cannot be an empty string`

File: sources/bamboohr-source/resources/spec.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      },
+      "domain": {
+        "type": "string",
+        "title": "domain",
+        "description": "Domain or Company domain",
+        "airbyte_secret": true
+      },
+      "version": {
+        "type": "string",
+        "title": "Api version",
```
Comment (cjwooo) [inline sources/bamboohr-source/resources/spec.json:27]: Capitalize `API`

File: sources/bamboohr-source/src/bamboohr.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      } catch (wrapError: any) {
+        errorMessage += wrapError.message;
+      }
+      throw new VError(errorMessage);
+    }
+  }
+
+  async *getUsers(): AsyncGenerator<User> {
+    try {
+      const users = await this.httpClient.get<any>(`/meta/users`);
```
Comment (cjwooo) [inline sources/bamboohr-source/src/bamboohr.ts:70]: Does this need to be paginated?

File: sources/buildkite-source/test/index.test.ts [hunk 4]
```diff
@@ -61,23 +61,23 @@ describe('index', () => {
...
-      false,
-      new VError(
-        "Please verify your token is correct. Error: Cannot read property 'get' of null"
-      ),
-    ]);
-  });
+  // test('check connection - incorrect config', async () => {
+  //   Buildkite.instance = jest.fn().mockImplementation(() => {
+  //     return new Buildkite(null, null, new Date('2010-03-27T14:03:51-0800'));
+  //   });
```
Comment (cjwooo) [inline sources/buildkite-source/test/index.test.ts:67]: Why is this commented out?

Comment (huongtn) [reply sources/buildkite-source/test/index.test.ts:67]: I uncommented

File: sources/bamboohr-source/resources/spec.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        "description": "BambooHR API Version",
+        "default": "v1"
+      }
```
[the remark this answers is not in the document]
Comment (huongtn) [reply sources/bamboohr-source/resources/spec.json:30]: yes

File: destinations/airbyte-faros-destination/src/converters/bamboohr/users.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+          name: user.department,
+        },
+      });
+    }
+
+    res.push({
+      model: 'identity_Identity',
+      record: {
+        uid,
+        fullName: user.fullName1,
```
[the remark this answers is not in the document]
Comment (huongtn) [reply destinations/airbyte-faros-destination/src/converters/bamboohr/users.ts:51]: ok

**Review by cjwooo — APPROVED** [review]

**Pull request merged**